EIPS/eip-3337.md

```diff
@@ -0,0 +1,99 @@
+
+In the current EVM model, a compiler wishing to use this pattern would have to store the frame pointer - which points to the start or end of the current memory stack frame - in memory, and load it each time they wish to reference it. For example, loading a value from memory offset by the frame pointer would require the following sequence of operations:
+
+| Opcode    | Gas used |
+|-----------|----------|
+| `PUSHn x` | 3        |
+| `PUSH1 0` | 3        |
+| `MLOAD`   | 3        |
+| `ADD`     | 3        |
+| `MLOAD`   | 3        |
+
+This consumes a total of 15 gas, and takes up at least 7 bytes of bytecode each time it is referenced. In contrast, after this EIP, the equivalent sequence of operations is:
+
+| Opcode    | Gas used |
+|-----------|----------|
+| `PUSH1 x` | 3        |
+| `MLOADFP` | 3        |
+
+This consumes only 6 gas, and takes at least 3 bytes of bytecode. The effort required from the EVM implementation is equivalent, costing only one extra addition operation over a regular `MLOAD`. The alternative of storing values on the stack, which requires 3 gas and 1 byte of bytecode for a `DUPn` operation, but it is now at most twice as efficient rather than 5 times as efficient, making storing values in memory a viable alternative.
+
+Likewise, before this EIP a frame-pointer relative store requires the following sequence of operations:
+| Opcode    | Gas used |
+|-----------|----------|
+| `PUSHn x` | 3        |
+| `PUSH1 0` | 3        |
+| `MLOAD`   | 3        |
+| `ADD`     | 3        |
+| `MSTORE`  | 3        |
+
+This consumes 15 gas and at least 7 bytes of bytecode. After this EIP, the equivalent sequence of operations is:
+
+| Opcode    | Gas used |
+|-----------|----------|
+| `PUSHn x` | 3        |
+| `MSTOREFP`| 3        |
+
+Consuming only 6 gas and at least 3 bytes of bytecode, while once again only requiring EVM implementations to do one extra addition operation. The alternative of storing values on the stack requires 6 gas and 2 bytes of bytecode for the sequence `SWAPn POP`, making it no more efficient than memory storage.
+
+## Specification
+### Parameters
+
+| Constant | Value |
+| - | - |
+| `FORK_BLOCK` | TBD |
+
+For blocks where `block.number >= FORK_BLOCK`, the following changes apply.
+
+### Frame pointer
+A new EVM internal state variable called the "frame pointer" is introduced. This is a signed integer that starts at 0.
```

Comment:
Probably needs to be specified as 256-bit twos-complement for the sake of the GETFP and SETFP operations.

Reply:
You may want to bring this up in the discussions-to to make sure it gets seen.  I agree that this needs more specificity!

Side note: We probably don't need 256-bits, likely 32-bits would be way more than enough?